src/qml/CoordinateLocator.qml

```diff
@@ -411,32 +480,38 @@ Item {
    * @param {number} angleDegrees - angle of the line in degrees.
    * @param {boolean} isRelativeAngle - whether the angle should be calculated relative to the last geometry segment
    * @param {number} screenSize - size of the screen. Used to make sure the end of the line is outside the screen.
+   * @param {number} forwardMode - true: aimed towards first point and false: snap to previous point
    */
-  function calculateSnapToAngleLineEndCoords(currentPoint, angleDegrees, isRelativeAngle, screenSize) {
+  function calculateSnapToAngleLineEndCoords(currentPoint, angleDegrees, isRelativeAngle, screenSize, forwardMode = false) {
     if (rubberbandModel == null) {
       return {};
     }
     const rubberbandPointsCount = rubberbandModel.vertexCount;
 
     // if the angle is null or undefined, return empty coordinate map
-    if (angleDegrees == null) {
+    if (angleDegrees === null) {
```

Comment:
This change is dangerous unless you fix the comment above. `== null` will check for both `null` and `undefined`, now only checks for `null`.